### PR TITLE
feat: use global components from altinn components

### PR DESCRIPTION
**Pull request by seanes**
## Related Issue(s)

- #{issue number}

### Dokumentasjon / Storybook / testdekning

- [ ] Dokumentasjon er oppdatert eller ikke relevant / nødvendig.
- [ ] Ny komponent har en eller flere `stories` i `Storybook`, eller så er ikke dette relevant.
- [ ] Det er blitt lagt til nye tester / eksiterende tester er blitt utvidet, eller tester er ikke relevant.

## Release Notes

- **New Features**
	- Introduced new `useAccounts`, `useFooter`, and `useGlobalMenu` hooks for enhanced account and menu management.
	- Added a `Background` component for dynamic styling based on props.
	- Implemented `useSearchDialogs` and `useSearchString` hooks to streamline search functionality.

- **Improvements**
	- Enhanced localization support with new keys in English and Norwegian.
	- Updated styling and layout management across various components for better responsiveness.

- **Bug Fixes**
	- Refined logic for handling search parameters and dialog retrieval to improve user experience.

- **Deprecations**
	- Removed several components including `Footer`, `Header`, and `Sidebar`, along with their associated styles and stories.

- **Documentation**
	- Updated documentation for automated testing strategies to clarify methodologies and best practices.

### CODE REVIEW

**Comment by mbacherycz** [comment]
I see some missing types in some places ex in PageLayout.tsx: 
Screenshot 2024-12-06 at 10 08 41

**Review by mbacherycz — APPROVED** [review]
Tons of work! Very good job 💪 
I left couple of suggestions for cleaner code and optimization. Let me know what you thing about them 👍

**Comment by mbacherycz** [comment]
We can easily find where TS is throwing errors (types). It prevents docker to start the app locally. 
Screenshot 2024-12-06 at 10 52 03

File: packages/frontend/src/components/PageLayout/GlobalMenu/useGlobalMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    {
+      id: 'chat',
+      groupId: 'global',
+      icon: 'person-chat',
+      title: t('menuBar.chat'),
+      size: 'lg',
+      as: (props: MenuItemProps) => <Link {...props} to="https://info.altinn.no/hjelp/" target="__blank" />,
+    },
+  ];
+  const sidebar: MenuItemProps[] = [
```
[the remark this answers is not in the document]
Comment (seanes) [reply packages/frontend/src/components/PageLayout/GlobalMenu/useGlobalMenu.tsx:64]: What do you think about this refactoring in updated solution?

[the remark this answers is not in the document]
Comment (mbacherycz) [reply packages/frontend/src/components/PageLayout/GlobalMenu/useGlobalMenu.tsx:64]: Looking good! 💯

File: packages/frontend/src/components/PageLayout/PageLayout.tsx [hunk 2]
```diff
@@ -87,29 +32,110 @@ export const ProtectedPageLayout = () => {
...
     queryClient.setQueryData(['search'], () => searchString || '');
-  });
+  }, [searchParams]);
+
+  const headerProps: HeaderProps = {
+    currentAccount: selectedAccount,
+    logo: {
+      as: (props: MenuItemProps) => <Link to="/" {...props} />,
+    },
+    badge:
```
[the remark this answers is not in the document]
Comment (seanes) [reply packages/frontend/src/components/PageLayout/PageLayout.tsx:74]: yup, I agree

**Review by alexdigdir — APPROVED** [review]

**Pull request merged**